measures/add_demo_noon_to_six/measure.py

```diff
@@ -0,0 +1,122 @@
+
+    def modeler_description(self):
+        """Human readable description of modeling approach.
+
+        The modeler description is intended for the energy modeler using the measure.
+        It should explain the measure's intent, and include any requirements about
+        how the baseline model must be set up, major assumptions made by the measure,
+        and relevant citations or references to applicable modeling resources
+        """
+        return "Add the demo Python plugin that runs a TES system (via a schedule) from noon until 6PM."
+
+    def arguments(self, workspace: openstudio.Workspace):
+        """Prepares user arguments for the measure.
+
+        Measure arguments define which -- if any -- input parameters the user may set before running the measure.
+        """
+        args = openstudio.measure.OSArgumentVector()
+
+        types = ['ThermalTank-Ice', 'ThermalTank-ChilledWater', 'PackagedIceStorage']
+        arg = openstudio.measure.OSArgument.makeChoiceArgument("tes_type", types, True)
+        arg.setDisplayName("TES type")
+        arg.setDescription("The TES system that is to be simulated.")
+        arg.setDefaultValue('ThermalTank-Ice')
+        args.append(arg)
+
+        arg = openstudio.measure.OSArgument.makeStringArgument("plugin_directory", True)
+        arg.setDisplayName("Python plugin directory")
+        arg.setDescription("A directory that is safe to write files to and is in the Python plugin search path list.")
+        arg.setDefaultValue('.')
+        args.append(arg)
+
+        return args
+
+    def run(
+        self,
+        workspace: openstudio.Workspace,
+        runner: openstudio.measure.OSRunner,
+        user_arguments: openstudio.measure.OSArgumentMap,
+    ):
+        """Defines what happens when the measure is run."""
+        super().run(workspace, runner, user_arguments)  # Do **NOT** remove this line
+
+        if not (runner.validateUserArguments(self.arguments(workspace), user_arguments)):
+            return False
+
+        # assign the user inputs to variables
+        tes_type = runner.getStringArgumentValue("tes_type", user_arguments)
+        plugin_directory = runner.getStringArgumentValue("plugin_directory", user_arguments)
+
+        # check the args for reasonableness
+        if not os.path.exists(plugin_directory):
+            runner.registerError(f'Output directory "{plugin_directory}" does not exist.')
+            return False
+
+        # report initial condition of model
+        runner.registerInitialCondition(f'Applying control schedule for "{tes_type}".')
+
+        # write out the system-specific details
+        if tes_type in ['ThermalTank-Ice', 'ThermalTank-ChilledWater']:
+            txt = thermaltank_custom_module
+        else:
+            pass
+
+        with open(os.path.join(plugin_directory, 'case_details.py'), 'w') as fp:
+            fp.write(txt)
```

Comment:
This line is writing out the case-specific information to a directory in the plugin search path.